spec/requests/restaurants_spec.rb

```diff
@@ -0,0 +1,132 @@
+      get restaurant_path(subject.id)
+      expect(response).to have_http_status(200)
+    end
+
+    it 'returns :not_found status on invalid uuid' do
+      get restaurant_path('this-is-not-a-uuid')
+      expect(response).to have_http_status(:not_found)
+    end
+
+    it 'returns the correct record' do
+      get restaurant_path(subject.id)
+      restaurant_matcher(subject, response_body)
+    end
+  end
+
+  describe 'POST /restaurants' do
+    it 'returns http success' do
+      post '/restaurants', params: FactoryBot.attributes_for(:restaurant)
+      expect(response).to have_http_status(:created)
+    end
+
+    it 'creates new record in the database' do
+      post '/restaurants', params: FactoryBot.attributes_for(:restaurant)
+      uuid = response_body['id']
+      rest = Restaurant.find_by(id: uuid)
+      expect(rest).to_not be_nil
+      restaurant_matcher(rest, response_body)
+    end
+
+    it 'returns error when params are invalid' do
+      invalid_attr = FactoryBot.attributes_for(:restaurant, :invalid_record)
+      post '/restaurants', params: invalid_attr
+      expect(response).to have_http_status(:unprocessable_entity)
+    end
+  end
+
+  describe 'PUT /restaurant/:id' do
+    subject { Restaurant.all.sample }
+
+    it 'returns https success' do
+      put restaurant_path(subject.id), params: FactoryBot.attributes_for(:restaurant)
+      expect(response).to have_http_status(:no_content)
+    end
+
+    it 'returns error when params are invalid' do
+      invalid_attr = FactoryBot.attributes_for(:restaurant, :invalid_record)
+      put restaurant_path(subject.id), params: invalid_attr
+      expect(response).to have_http_status(:unprocessable_entity)
+    end
+
+    it 'updates the record fields' do
+      new_rest = FactoryBot.build(:restaurant, id: subject.id)
+      put restaurant_path(subject.id), params: new_rest.attributes
+      rest = Restaurant.find(subject.id)
+      restaurant_matcher(rest, new_rest)
+    end
+  end
+
+  describe 'DELETE /restaurant/:id' do
+    subject { Restaurant.all.sample }
+
+    it 'returns http success' do
+      delete restaurant_path(subject.id)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'fails to find the record in DB' do
+      delete restaurant_path(subject.id)
+      expect(Restaurant.find_by(id: subject.id)).to be_nil
+    end
+  end
+
+  private
+
+  def response_body
+    JSON.parse(response.body)
+  end
+
+  def restaurant_matcher(rest_ref, rest_sample)
+    expect(rest_sample['name']).to eq(rest_ref.name)
+    expect(rest_sample['cuisine']).to eq(rest_ref.cuisine.to_s)
+    expect(rest_sample['id']).to eq(rest_ref.id)
+    expect(rest_sample['address']).to eq(rest_ref.address)
+    expect(rest_sample['max_delivery_time_min']).to eq(rest_ref.max_delivery_time_min)
+    float_matcher(rest_sample['latitude'], rest_ref.latitude)
+    float_matcher(rest_sample['longitude'], rest_ref.longitude)
+    float_matcher(rest_sample['rating'], rest_ref.rating)
```

Comment:
if one of the values is nil or not exist, is it still work?

Reply:
yes as long as they both are nil and not only one of them,
for not-exist I added a matcher to check that they have all the attributes